### DISCUSSION TITLE
Unmarshalling question

### DISCUSSION THREAD

**Discussion by RaulBernal** [cosmos/community#8] [Chain Builders (Cosmos SDK)] [answered]
Hello, we are finishing the v0.45 -> v0.46 migration at BitCanna.
I was wondering if this way of processing the data is correct for v0.46?
- https://github.com/BitCannaGlobal/bcna/blob/6a6e18e21a7c6187f76d23bcc7fada4ce8f1ae4d/x/bcna/keeper/bitcannaid.go#L47
- https://github.com/BitCannaGlobal/bcna/blob/6a6e18e21a7c6187f76d23bcc7fada4ce8f1ae4d/x/bcna/keeper/bitcannaid.go#L59

I guess yes because we use our own codec but the asking comes checking this changes:
- https://github.com/ingenuity-build/quicksilver/pull/304/files#diff-d86d338767739d0d60c746925ca8a9087e7e403d1d01cb92532ec4206ebe4195L117

**Comment by RaulBernal** [comment]
I should say tests are ok
<img width="350" alt="image" src="https://user-images.githubusercontent.com/3751926/227491700-4ef63802-214a-4670-bd1f-c8deded35f88.png">

**Reply by RaulBernal** [reply]
@julienrbrt Do you realize why when I ask for list all the `bitcannaid` registries by LCD it doesn't affect the `?height=xxxxx` filter?
I solved for the CLI (https://github.com/BitCannaGlobal/bcna/commit/2e90dc3541eb0f0ebedaa5154e87d8faf3a6c771) but can't realize where to fix for LCD.
I guess something is missing with the context here, in example:
https://github.com/BitCannaGlobal/bcna/blob/main/x/bcna/keeper/bitcannaid.go#L80

**Reply by RaulBernal** [reply]
I saw a lot of custom modules in the ecosystem for different chains lacking this issue. 
In example, the liquidity module for gaiad also do: 
`gaiad q liquidity pools --height 1`  produces the same output than `gaiad q liquidity pools --height 10638712` than `gaiad q liquidity pools` 
  The same counts for the LCD. I know how to fix the CLI here https://github.com/BitCannaGlobal/bcna/blob/4d50e3da1f53363f0791cab885dd64892b70129d/x/bcna/client/cli/query_bitcannaid.go#L19
but not the LCD reply.

Issue created [here](https://github.com/Gravity-Devs/liquidity/issues/28)

**Comment by tac0turtle** [comment, marked as answer]
the change you linked in quicksilver seems they had a type within the message using an Any which in our case needs to be deconstructed. But then they realised its unneeded. If you dont have to unpack any then proto.* is safe. 

BTW, thank you so much for asking this is question in the org level discussion board!

**Reply by RaulBernal** [reply]
Thanks for your reply, it light my concern (_If you dont have to unpack any then proto.* is safe_ )
